model/riscv_pmp_regs.sail

```diff
@@ -164,61 +157,59 @@ function pmpLocked(cfg: Pmpcfg_ent) -> bool =
 function pmpTORLocked(cfg: Pmpcfg_ent) -> bool =
    (cfg.L() == 0b1) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
 
-function pmpWriteCfg(cfg: Pmpcfg_ent, v: bits(8)) -> Pmpcfg_ent =
-  if pmpLocked(cfg) then cfg
-  else Mk_Pmpcfg_ent(v & 0x9f)  // Bits 5 and 6 are zero.
+function pmpWriteCfg(n: range(0, 63), cfg: Pmpcfg_ent, v: bits(8)) -> Pmpcfg_ent =
+  // If the PMP entry is locked or it is not writable (all PMP entry fields
+  // are WARL and can be read-only zero) then don't modify it.
+  if pmpLocked(cfg) | sys_pmp_writable()[n] == bitzero then cfg
+  else {
+    // Bits 5 and 6 are zero.
+    let cfg = Mk_Pmpcfg_ent(v & 0x9f);
+
+    // "The R, W, and X fields form a collective WARL field for which the combinations with R=0 and W=1 are reserved."
+    // In this implementation if R=0 and W=1 then R, W and X are all set to 0.
+    // This is the least risky option from a security perspective.
```

Comment:
Correct,  but the spec doesn't require that. It needs user specified rules (and the Epmp entry explicitly allows that combination for shared rule